packages/block-library/src/block/edit.js

```diff
@@ -314,78 +178,25 @@ function ReusableBlockEdit( {
 	useEffect( () => {
 		setBlockEditMode(
 			setBlockEditingMode,
-			innerBlocks,
+			blocks,
 			// Disable editing if the pattern itself is disabled.
 			editingMode === 'disabled' || ! hasPatternOverridesSource
 				? 'disabled'
 				: undefined
 		);
 	}, [
 		editingMode,
-		innerBlocks,
+		blocks,
 		setBlockEditingMode,
 		hasPatternOverridesSource,
 	] );
 
 	const canOverrideBlocks = useMemo(
-		() => hasPatternOverridesSource && hasOverridableBlocks( innerBlocks ),
-		[ hasPatternOverridesSource, innerBlocks ]
+		() => hasPatternOverridesSource && hasOverridableBlocks( blocks ),
+		[ hasPatternOverridesSource, blocks ]
 	);
 
-	const initialBlocks = useMemo(
-		() =>
-			// Clone the blocks to generate new client IDs.
-			editedRecord.blocks?.map( ( block ) => cloneBlock( block ) ) ??
-			( editedRecord.content && typeof editedRecord.content !== 'function'
-				? parse( editedRecord.content )
-				: [] ),
-		[ editedRecord.blocks, editedRecord.content ]
-	);
-
-	const legacyIdMap = useRef( {} );
-
-	// Apply the initial overrides from the pattern block to the inner blocks.
-	useEffect( () => {
-		// Build a map of clientIds to the old nano id system to provide back compat.
-		legacyIdMap.current = getLegacyIdMap(
-			initialBlocks,
-			initialContent.current
-		);
-		defaultContent.current = {};
-		const originalEditingMode = getBlockEditingMode( patternClientId );
-		// Replace the contents of the blocks with the overrides.
-		registry.batch( () => {
-			setBlockEditingMode( patternClientId, 'default' );
-			syncDerivedUpdates( () => {
-				const blocks = hasPatternOverridesSource
-					? applyInitialContentValuesToInnerBlocks(
-							initialBlocks,
-							initialContent.current,
-							defaultContent.current,
-							legacyIdMap.current
-					  )
-					: initialBlocks;
-
-				replaceInnerBlocks( patternClientId, blocks );
-			} );
-			setBlockEditingMode( patternClientId, originalEditingMode );
```

Comment:
The fact that we were making these calls here in trunk could have "hidden" the race conditions. It doesn't mean the race conditions didn't exist on trunk, they did, but they were hidden or maybe they resulted in different bugs (like things being editable in the template while in "template-lock" mode)